napari/view_layers.py

```diff
@@ -146,15 +149,42 @@ def _generate_view_function(layer_string: str, method_name: str = None):
     view_func.__doc__ = merge_docs(add_method, layer_string)
 
 
-for _layer in (
-    'image',
-    'points',
-    'labels',
-    'shapes',
-    'surface',
-    'vectors',
-    'tracks',
-):
+for _layer in layers.NAMES:
     _generate_view_function(_layer)
 
 _generate_view_function('path', 'open')
+
+
+def _generate_stubs(output=__file__.replace(".py", ".pyi")):
```

Comment:
how come `_generate_cls_stubs` is in `napari.utils.misc` and this is not?

Reply:
this one was a bit hard to make generic...  but I should have tried harder :)  I can make it `generate_module_stubs` and have it accept the names that should be added